Projects/Features/PlaylistFeature/Sources/Reactors/PlaylistDetailReactor.swift

```diff
@@ -0,0 +1,220 @@
+import AuthDomainInterface
+import BaseDomainInterface
+import BaseFeature
+import ErrorModule
+import Foundation
+import PlayListDomainInterface
+import ReactorKit
+import RxCocoa
+import RxRelay
+import RxSwift
+import SongsDomainInterface
+import Utility
+
+#warning("저장하기, 업데이트")
+
+internal final class PlaylistDetailReactor: Reactor {
+    public enum Action {
+        case viewDidLoad
+        case itemMoved(ItemMovedEvent)
+        case tapEdit
+        case completeEdit
+        case tapSong(Int)
+        case tapAll(isSelecting: Bool)
+        case undo
+    }
+
+    public enum Mutation {
+        case fetchData(PlaylistMetaData)
+        case updateOrder([SongEntity])
+        case beginEditing
+        case save
+        case changeSelectedState((data: [SongEntity], selectedCount: Int))
+        case changeAllState((data: [SongEntity], selectedCount: Int))
+        case undo
+    }
+
+    public struct State {
+        var dataSource: [PlayListDetailSectionModel]
+        var backupDataSource: [PlayListDetailSectionModel]
+        var header: PlayListHeaderModel
+        var selectedItemCount: Int
+        var isEditing: Bool
+    }
+
+    public var initialState: State
+    public let type: PlayListType!
+    public let key: String!
```

Comment:
```suggestion
    public let type: PlayListType
    public let key: String
```
implicitly unwrapping optional로 둘 필요 없을거같아요